File: runtime/Jlib/belong.java
package bigloo;

public class belong extends numeral
{
  public final long value;

  public belong( final long  value )
  {
    this.value= value;
  }

  public static belong make_elong( final String  s )
  {
    return new belong( Long.parseLong( s ) );
  }

  public void write( final output_port  p )
  {
    p.write( "#e" + value );
  }
}

File: runtime/CSlib/belong.cs
using System;

namespace bigloo
{
  public sealed class belong: numeral
  {
    public readonly long value;

    public belong( long  value )
    {
      this.value= value;
    }

    public static belong make_elong( String  s )
    {
      return new belong( Int64.Parse( s ) );
    }

    public override void write( output_port  p )
    {
      p.write( "#e" + value );
    }
  }
}
